### PR TITLE
Added release notes for 3.21.0

### PR DESCRIPTION
**Pull request by robfalck**
### Summary

Adds release notes for the OpenMDAO 3.21.0 release

### Related Issues

- None

### Backwards incompatibilities

- None

### New Dependencies

- None

### CODE REVIEW

File: release_notes.md [hunk 1]
```diff
@@ -1,3 +1,62 @@
...
+- **POEM_073**: Allow DOEDriver to record derivatives to allow user to check derivatives across model. [#2648](https://github.com/OpenMDAO/OpenMDAO/pull/2648)
+- Added optional arguments `method` and `excludes` to `assert_no_approx_partials` [#2652](https://github.com/OpenMDAO/OpenMDAO/pull/2652)
+
+## Bug Fixes
+
+- Accounted for y-intercept on bounds in linear constraints given to pyoptsparse. [#2605](https://github.com/OpenMDAO/OpenMDAO/pull/2605)
+- Fixed bug where models with discrete connections were reconfigured between setup calls. [#2610](https://github.com/OpenMDAO/OpenMDAO/pull/2610)
+- Added a better error message when DirectSolver fails and broadened conditions to check for rank deficiency. [#2614](https://github.com/OpenMDAO/OpenMDAO/pull/2614)
+- Sorted information for N2 generation to ensure consistency in created diagrams and fixed an error when drawing cycle arrows. [#2616](https://github.com/OpenMDAO/OpenMDAO/pull/2616)
+- Fixed file save dialongs for the N2 diagram. [#2619](https://github.com/OpenMDAO/OpenMDAO/pull/2619)
```
Comment (swryan) [inline release_notes.md:38]: typo: dialongs

File: release_notes.md [hunk 2]
```diff
@@ -1,3 +1,62 @@
...
+- Fixed a bug in check_totals when an aliased constraint is defined in a child system. [#2621](https://github.com/OpenMDAO/OpenMDAO/pull/2621)
+- Fixed the error message to be more helpful when a user calls `Problem.list_problem_vars()` before `final_setup`. [#2624](https://github.com/OpenMDAO/OpenMDAO/pull/2624)
+- Removed deprecation warnings under python 3.10 when entry points are used. [#2625](https://github.com/OpenMDAO/OpenMDAO/pull/2625)
+- Fixed failure in load_case when variable is distributed. [#2627](https://github.com/OpenMDAO/OpenMDAO/pull/2627)
+- Updated check for when coloring doesn't meet min improvement percentage to prevent errant messages. [#2630](https://github.com/OpenMDAO/OpenMDAO/pull/2630)
```
Comment (swryan) [inline release_notes.md:43]: suggest: min -> minimum

File: release_notes.md [hunk 3]
```diff
@@ -1,3 +1,62 @@
...
+- Fixed to allow a shape mismatch in AutoIVC if all promoted inputs have different shapes but unambiguous storage orders. [#2632](https://github.com/OpenMDAO/OpenMDAO/pull/2632)
+- Fixed a bug in `AddSubtractComp` when reusing an input defined in a previous equation. [#2636](https://github.com/OpenMDAO/OpenMDAO/pull/2636)
+- pip install myst_nb in the docs when run on Colab. [#2638](https://github.com/OpenMDAO/OpenMDAO/pull/2638)
```
Comment (swryan) [inline release_notes.md:46]: suggest: include `myst_nb` with optional `notebooks` dependencies to support Google Colab

File: release_notes.md [hunk 4]
```diff
@@ -1,3 +1,62 @@
...
+- Fix for CaseViewer issue caused by matplotlib change. [#2640](https://github.com/OpenMDAO/OpenMDAO/pull/2640)
+- Fix to allow "OPENMDAO_REPORTS=none" to diable reports, per the docs. [#2649](https://github.com/OpenMDAO/OpenMDAO/pull/2649)
```
Comment (swryan) [inline release_notes.md:48]: typo: diable

File: release_notes.md [hunk 5]
```diff
@@ -1,3 +1,62 @@
...
+
+## Miscellaneous
+
+- Added `long_description_content_type` to setup.py for rendering on pypi. [#2604](https://github.com/OpenMDAO/OpenMDAO/pull/2604)
+- Updated `Sellar` model used for testing and documentation.. [#2613](https://github.com/OpenMDAO/OpenMDAO/pull/2613)
+- Added coverage improvements. [#2615](https://github.com/OpenMDAO/OpenMDAO/pull/2615)
+- Due to issues with conda switch github workflow to use mamba and pre-install some jupyter dependencies. [#2618](https://github.com/OpenMDAO/OpenMDAO/pull/2618)
```
Comment (swryan) [inline release_notes.md:55]: suggest: comma after conda for readability

**Review by swryan — APPROVED** [review]

**Review by Kenneth-T-Moore — APPROVED** [review]

**Pull request merged**